### DISCUSSION TITLE
Devtools for Native Environment

### DISCUSSION THREAD

**Discussion by heywhy** [vuejs/pinia#2281] [Ideas]
There is an ongoing effort to allow NativescriptVue to utilize the standalone devtools. The current approach taken by Pinia to setup a devtool plugin is limiting in the sense that the `IS_CLIENT` expects the `window` variable to exists, and this isn't applicable to environments like Nativescript.

So, I will suggest that the existence of `__VUE_DEVTOOLS_GLOBAL_HOOK__` should override the flag.

```js
const USE_DEVTOOLS = ((process.env.NODE_ENV !== 'production') || (typeof __VUE_PROD_DEVTOOLS__ !== 'undefined' && __VUE_PROD_DEVTOOLS__)) && !(process.env.NODE_ENV === 'test') && !!__VUE_DEVTOOLS_GLOBAL_HOOK__;
```

**Comment by posva** [comment]
Sorry for the late reply. I wonder what's the state of this today. Feel free to open a new discussion with up to date info and to ping nativescript contributors. I suspect that pinia is not the only lib with these issues
